### PR TITLE
handle remote type create git remote resource first pass

### PR DESCRIPTION
**Pull request by kdai7**

### CODE REVIEW

File: lib/remoteResource.js [hunk 1]
```diff
@@ -26,16 +26,71 @@ const requestsTemplate = `{
...
+        sub.remote.parameters.map((param) => {
+          if (param.key == 'repo') {
+            repo = param.value;
+          } else if (param.key == 'branch') {
+            branch = param.value;
+          } else if (param.key == 'filePath') {
+            filePath = param.value;
+          } else if (param.key == 'secretName') {
+            secretNameGit = param.value;
+          }
```
Comment (carrolp) [inline lib/remoteResource.js:88]:
```suggestion
        for( const param of sub.remote.parameters ) {
          if (param.key == 'repo') {
            repo = param.value;
          } else if (param.key == 'branch') {
            branch = param.value;
          } else if (param.key == 'filePath') {
            filePath = param.value;
          } else if (param.key == 'secretName') {
            secretNameGit = param.value;
          }
```
`map()` isn't exactly the right approach here ("The map() method creates a new array populated with the results of calling a provided function on every element in the calling array").  Instead, try `sub.remote.parameters.forEach(...` or (as in suggestion above) `for( const param of sub.remote.parameters ) {...}`.

File: lib/remoteResource.js [hunk 2]
```diff
@@ -26,16 +26,71 @@ const requestsTemplate = `{
...
       const secretName = `clustersubscription-${sub.subscriptionUuid}-secret`;
-      const rendered = Mustache.render(requestsTemplate, { url: url, secretName: secretName, namespace: NAMESPACE });
+      let rendered;
+      let backendService;
+      let secretNameGit = 'satellite-config-gitops';
+      if (sub.url) {
+        rendered = Mustache.render(requestsTemplate, { url: url, secretName: secretName, namespace: NAMESPACE });
+        backendService = 'generic';
+      } else if (sub.remote) {
+        const provider = sub.remote.remoteType;
```
Comment (carrolp) [inline lib/remoteResource.js:75]: Does the remoteType value need to be checked to differentiate between `github` and `gitlab` or an unexpected value?  Similarly, does the `backendService = 'git'` need to differentiate between `github` and `gitlab` or other?

File: lib/remoteResource.js (hunk 1 above)
Comment (adamkingit) [reply lib/remoteResource.js:88]: .foreach(... is a good alternative

File: lib/remoteResource.js [hunk 3]
```diff
@@ -26,16 +26,71 @@ const requestsTemplate = `{
...
   try {
     return Promise.all(subscriptions.map(async sub => {
       const apiKeyBase64 = Buffer.from(apiKey).toString('base64');
       const url = `${razeeApi}/${sub.url}`;
       const secretName = `clustersubscription-${sub.subscriptionUuid}-secret`;
-      const rendered = Mustache.render(requestsTemplate, { url: url, secretName: secretName, namespace: NAMESPACE });
+      let rendered;
+      let backendService;
+      let secretNameGit = 'satellite-config-gitops';
+      if (sub.url) {
```
Comment (adamkingit) [inline lib/remoteResource.js:71]: Try changing this into a factory to simplify adding additional remoterespource backends such as S3 in the future.

File: lib/remoteResource.js (hunk 2 above)
Comment (kdai7) [reply lib/remoteResource.js]: backendservice doesn't need to differentiate, rr will handle it

File: lib/remote/git.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        }
+        
+      }
+    };
+    const requestsTemplate = JSON.stringify(requests);
+    return Mustache.render(requestsTemplate, { repo: this._repo, provider: this.provider, branch: this._branch, filePath: this._filePath, secretName: this.secretName, secretNameGit: this.secretNameGit, namespace: this.namespace});
+  }
+
+
+};
```
Comment (adamkingit) [inline lib/remote/git.js:79]:
```suggestion
};

```

File: lib/remote/remote.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  get secretNameGit() {
+    return this._secretNameGit;
+  }
+
+  get parameters() {
+    return this._parameters;
+  }
+  
+};
```
Comment (adamkingit) [inline lib/remote/remote.js:46]:
```suggestion
};

```

File: lib/remoteResource.js (hunk 3 above)
Comment (adamkingit) [reply lib/remoteResource.js:40]: Thinking you can drive your factory off the backendService value

File: lib/backendservice/generic.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+              }
+            }
+          }
+        }
+      }
+    };
+    const requestsTemplate = JSON.stringify(requests);
+    return Mustache.render(requestsTemplate, { url: this.url, secretName: this.secretName, namespace: this.namespace });
+  }
+};
```
Comment (adamkingit) [inline lib/backendservice/generic.js:50]:
```suggestion
};

```

**Review by adamkingit — APPROVED** [review]

**Pull request merged**